manager/controllers/app/np_isolation_read_flow_test.go

```diff
@@ -0,0 +1,376 @@
+// Copyright 2021 IBM Corp.
+// SPDX-License-Identifier: Apache-2.0
+
+package app
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/aws/aws-sdk-go/service/s3/s3manager"
+	. "github.com/onsi/ginkgo/v2"
+	"github.com/onsi/gomega"
+	"gopkg.in/yaml.v2"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/client-go/kubernetes/scheme"
+	restclient "k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/remotecommand"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/apiutil"
+
+	fapp "fybrik.io/fybrik/manager/apis/app/v1beta1"
+)
+
+func execCmdCommand(restClient restclient.Interface, config *restclient.Config, namespace string, command string,
+	stdout io.Writer, stderr io.Writer) error {
+	cmd := []string{
+		"sh",
+		"-c",
+		command,
+	}
+	req := restClient.Post().Resource("pods").Name("my-shell").
+		Namespace(namespace).SubResource("exec")
+	option := &v1.PodExecOptions{
+		Command: cmd,
+		Stdin:   false,
+		Stdout:  true,
+		Stderr:  true,
+		TTY:     true,
+	}
+	req.VersionedParams(
+		option,
+		scheme.ParameterCodec,
+	)
+	exec, err := remotecommand.NewSPDYExecutor(config, "POST", req.URL())
+	if err != nil {
+		return err
+	}
+	err = exec.Stream(remotecommand.StreamOptions{
+		Stdin:  nil,
+		Stdout: stdout,
+		Stderr: stderr,
+	})
+
+	return err
+}
+
+func uploadToS3(endpoint string, g gomega.Gomega) {
+	region := "theshire"
+	bucket := "bucket1"
+	key1 := "data.csv"
+	filename := "../../testdata/data.csv"
+	s3credentials := credentials.NewStaticCredentials("ak", "sk", "")
+	sess2 := session.Must(session.NewSession(&aws.Config{
+		Credentials:      s3credentials,
+		Endpoint:         &endpoint,
+		Region:           &region,
+		S3ForcePathStyle: aws.Bool(true),
+	}))
+	s3Client2 := s3.New(sess2)
```

Comment:
why do you call all objects with suffix "2" ?

Reply:
Fixed.